test/e2e/node/node_problem_detector.go

```diff
@@ -60,18 +61,30 @@ var _ = SIGDescribe("NodeProblemDetector [DisabledForLargeClusters]", func() {
 		e2eskipper.SkipUnlessSSHKeyPresent()
 
 		ginkgo.By("Getting all nodes and their SSH-able IP addresses")
-		nodes, err := e2enode.GetReadySchedulableNodes(f.ClientSet)
+		readyNodes, err := e2enode.GetReadySchedulableNodes(f.ClientSet)
 		framework.ExpectNoError(err)
+
+		nodes := []v1.Node{}
 		hosts := []string{}
-		for _, node := range nodes.Items {
+		for _, node := range readyNodes.Items {
+			host := ""
 			for _, addr := range node.Status.Addresses {
 				if addr.Type == v1.NodeExternalIP {
-					hosts = append(hosts, net.JoinHostPort(addr.Address, "22"))
+					host = net.JoinHostPort(addr.Address, "22")
 					break
 				}
 			}
+			// Not every node has to have an external IP address.
+			if len(host) > 0 {
+				nodes = append(nodes, node)
+				hosts = append(hosts, host)
+			}
+		}
+
+		if len(nodes) > maxNodesToProcess {
```

Comment:
should this check that we got at least one node in the list?

Reply:
/hold 

for this comment.

Reply:
> should this check that we got at least one node in the list?

What should we do if there isn't? Technically, it's a completely valid setup for a node to not have public IP. So in that case, this test is incompatible with that. But we won't fail the test anyway - we may potentially log something, but not sure how much it helps us...

Reply:
in case there is no valid IP test doesn't test anything. Maybe `ginkgo.Skip` as failed prerequisites?

Reply:
done